testing/generation/function_coverage/main.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/dolthub/go-mysql-server/sql"
+
+	_ "github.com/dolthub/doltgresql/server/functions"
+	"github.com/dolthub/doltgresql/server/functions/framework"
+	"github.com/dolthub/doltgresql/testing/generation/utils"
+)
+
+// Assertion represents a single SELECT assertion.
+type Assertion struct {
+	Stmt  string
+	Rows  []sql.Row
+	Error bool
+}
+
+func main() {
+	framework.Initialize()
+	rootFolder, err := utils.GetRootFolder()
+	if err != nil {
+		fmt.Printf("%s\n", err.Error())
+		return
+	}
+	rootFolder = rootFolder.MoveRoot("testing/generation/function_coverage")
+
+	// Sort the strings for determinism, since range iteration on maps is non-deterministic
+	functionNames := make([]string, 0, len(framework.Catalog))
+	for functionName := range framework.Catalog {
+		functionNames = append(functionNames, functionName)
+	}
+	sort.Strings(functionNames)
+
+	for _, functionName := range functionNames {
+		var assertions []Assertion
+
+		utils.PrintSectionMarker(functionName, '+', 80)
+
+	FunctionLoop:
+		for _, function := range framework.Catalog[functionName] {
+			var literalGeneratorParams []utils.StatementGenerator
+			literalGeneratorParams = append(literalGeneratorParams, utils.Text(functionName+"("))
+			for i, paramType := range function.GetParameters() {
+				if i > 0 {
+					literalGeneratorParams = append(literalGeneratorParams, utils.Text(", "))
+				}
+				if generator, ok := valueMappings[paramType.BaseID()]; ok {
+					literalGeneratorParams = append(literalGeneratorParams, generator)
+				} else {
+					fmt.Printf("missing support for functions with the parameter type: `%s`\n", paramType.String())
+					continue FunctionLoop
+				}
+			}
+			literalGeneratorParams = append(literalGeneratorParams, utils.Text(")"))
+			literalGenerator := utils.Collection(
+				utils.Text("SELECT"),
+				utils.Collection(literalGeneratorParams...),
+				utils.Text(";"),
+			)
+
+			randomInts, err := utils.GenerateRandomInts(500, literalGenerator.Permutations())
```

Comment:
Nope, and that's intentional. For example, say we have sine and it takes a float64. That's a ton of potential values we could test, so rather than have petabytes of test files, we randomly select 500 permutations of all possible tests (according to the generator). Every time a new set of tests are generated, we'll cycle to a different set, which will test different functionality. If our logic is right, then all possible permutations will work.

This is exactly the same thing that we do for `testing/generation/command_docs` for the exact same reason.